Remove py2 workarounds 1/2

terrdavis opened the pull request:
Breaking up #12092, per [discussion](https://github.com/ipython/ipython/pull/12092#issuecomment-580899602).

terrdavis committed:
Remove unused sys & fallback pathlib (3.5+) imports.

terrdavis committed:
Remove python 2 only import in IPython.core.ultratb

terrdavis committed:
Remove python 2 specific line.

terrdavis committed:
Remove python 2 references from docstrings, where appropriate.

terrdavis committed:
Remove python 2 only long and unicode type references.

Carreau commented on IPython/core/ultratb.py:
```diff
@@ -101,10 +101,7 @@
-    generate_tokens = tokenize.generate_tokens
-except AttributeError:  # Python 3
-    generate_tokens = tokenize.tokenize
+generate_tokens = tokenize.tokenize
```
The comments appear to be wrong, `generate_tokens = tokenize.generate_tokens` work on Python 3; and this is likely the cause for the  test failures.

Carreau committed:
Attempt to fix test failure with generate_tokens

It looks like the try/except comments were wrong and generate_tokens is available on Py3
